### PR TITLE
Add initial version of Identification form component

### PR DESCRIPTION
**Pull request by lucasecdb**
#### What problem is this solving?
Adds the initial version of the Identification form component.

[Related clubhouse story](https://app.clubhouse.io/vtex/story/32231/etapa-de-identifica%C3%A7%C3%A3o-do-usu%C3%A1rio).

#### How should this be manually tested?
[Workspace](https://chkio--checkoutio.myvtex.com/checkout/identification)

#### Checklist/Reminders

- [ ] Updated `README.md`.
- [x] Updated `CHANGELOG.md`.
- [x] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Screenshots or example usage
Screen Shot 2020-03-02 at 14 21 48

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: store/blocks.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "checkout-identification": {
+    "blocks": ["image#default"]
+  },
+  "image#default": {
+    "props": {
+      "src": "https://storecomponents.vtexassets.com/arquivos/store-theme-logo-mobile.png",
```
Comment (lucasecdb) [inline store/blocks.json:7]: this is a workaround until the devtools team resolves the issues of using the `assets` builder with a "long" app name (exceeds the vbase bucket name size of 50 characters)

File: messages/es.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "store/checkout-payment-step-title": "Pago",
+  "store/checkout-identification-presentation": "Por favor, ingrese su correo electónico",
```
Comment (darlenedms) [inline messages/es.json]:
```suggestion
  "store/checkout-identification-presentation": "Por favor, ingrese su correo electrónico",
```

File: messages/es.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "store/checkout-identification-secure-badge-title": "Seguro",
+  "store/checkout-identification-email-label": "Correo electónico",
```
Comment (darlenedms) [inline messages/es.json]:
```suggestion
  "store/checkout-identification-email-label": "Correo electrónico",
```

File: react/Identification.tsx [hunk 4]
```diff
@@ -1,7 +1,174 @@
...
+  title: {
+    id: 'store/checkout-identification-presentation',
+  },
+  secure: {
+    id: 'store/checkout-identification-secure-badge-title',
+  },
+  emailLabel: {
+    id: 'store/checkout-identification-email-label',
+  },
+  contineButtonLabel: {
```
Comment (darlenedms) [inline react/Identification.tsx]:
```suggestion
  continueButtonLabel: {
```

File: react/Identification.tsx [hunk 5]
```diff
@@ -1,7 +1,174 @@
...
+        onChange={handleEmailChange}
+        errorMessage={showError && intl.formatMessage(messages.invalidEmail)}
+      />
+
+      <div className="flex mt5">
+        <Button block disabled={loading} type="submit">
+          {loading ? (
+            <Spinner size={24} />
+          ) : (
+            <FormattedMessage {...messages.contineButtonLabel} />
```
Comment (darlenedms) [inline react/Identification.tsx]:
```suggestion
            <FormattedMessage {...messages.continueButtonLabel} />
```

File: react/package.json [hunk 6]
```diff
@@ -4,12 +4,21 @@
...
   },
   "devDependencies": {
+    "@types/classnames": "^2.2.9",
     "@types/graphql": "^14.5.0",
     "@types/jest": "^25.1.3",
     "@types/node": "^13.7.7",
-    "@types/react": "^16.9.23"
+    "@types/react": "^16.9.23",
+    "vtex.checkout-graphql": "https://lucas--checkoutio.myvtex.com/_v/private/typings/linked/v1/vtex.checkout-graphql@0.23.0+build1582915320/public/@types/vtex.checkout-graphql",
+    "vtex.checkout-resources": "https://lucas--checkoutio.myvtex.com/_v/private/typings/linked/v1/vtex.checkout-resources@0.19.0+build1582915328/public/@types/vtex.checkout-resources",
```
Comment (marcoskwkm) [inline react/package.json]: These depend on your workspace. You should either remove them or rerun `vtex setup` without no app linked.

**Review by darlenedms — APPROVED** [review]

File: react/typings/vtex.checkout-resources.d.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable import/no-duplicates, import/order */
+
+declare module 'vtex.checkout-resources/QueryProfile' {
+  import { DocumentNode } from 'graphql'
+
+  const query: DocumentNode
+  export default query
+}
```
Comment (marcoskwkm) [inline react/typings/vtex.checkout-resources.d.ts]: Could you test something for me? Try changing the extensions of the `checkout-resources` queries/mutations from `.js` to `.ts(x)` and see if their types are exported, so you won't need this file anymore.

File: react/package.json (hunk 6 above)
Comment (lucasecdb) [reply react/package.json]: just updated the packages!

File: react/typings/vtex.checkout-resources.d.ts (hunk 7 above)
Comment (lucasecdb) [reply react/typings/vtex.checkout-resources.d.ts]: I'll try!

Comment (lucasecdb) [reply react/typings/vtex.checkout-resources.d.ts]: done

File: react/Identification.tsx [hunk 8]
```diff
@@ -1,7 +1,174 @@
-import React from 'react'
+import classNames from 'classnames'
+import React, { useState, useEffect } from 'react'
+import { useLazyQuery, useMutation } from 'react-apollo'
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl'
+import { IconCheck, Input, Button, Divider, Spinner } from 'vtex.styleguide'
+import { useChildBlock, ExtensionPoint, useRuntime } from 'vtex.render-runtime'
+import ProfileQuery from 'vtex.checkout-resources/QueryProfile'
+import UpdateOrderFormProfileMutation from 'vtex.checkout-resources/MutationUpdateOrderFormProfile'
```
Comment (marcoskwkm) [inline react/Identification.tsx:8]: How about 
```suggestion
import { QueryProfile as ProfileQuery } from 'vtex.checkout-resources'
import { MutationUpdateOrderFormProfile as UpdateOrderFormProfileMutation } from 'vtex.checkout-resources'
```
and deleting `typings/vtex.checkout-resources.d.ts`?

Comment (lucasecdb) [reply react/Identification.tsx:8]: won't that result in the same issue we had of loading all the queries in runtime while using just a couple of them?

Comment (marcoskwkm) [reply react/Identification.tsx:8]: /O\\
You are right. It is just so bad having that `d.ts` file but I guess we don't have a better solution for now.

**Review by marcoskwkm — APPROVED** [review]

File: react/Identification.tsx (hunk 8 above)
Comment (lucasecdb) [reply react/Identification.tsx:8]: I don't really like maintaining that either, but that's a limitation in the package we use :-/

**Pull request merged**